// redux-form-website-template/es/components/CodeFund.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var id = '141';

var heights = {
  'image-only': 211,
  default: 130
};

var CodeFund = function (_React$Component) {
  _inherits(CodeFund, _React$Component);

  function CodeFund() {
    _classCallCheck(this, CodeFund);

    return _possibleConstructorReturn(this, (CodeFund.__proto__ || Object.getPrototypeOf(CodeFund)).apply(this, arguments));
  }

  _createClass(CodeFund, [{
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate() {
      return false;
    }
  }, {
    key: 'render',
    value: function render() {
      var template = this.props.template;

      var height = heights[template] || 150;
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement('div', {
          id: 'codefund',
          className: 'benevolent-sponsor template-' + template,
          style: { minHeight: height + 'px' }
        }),
        _react2.default.createElement('script', {
          src: 'https://codefund.app/properties/' + id + '/funder.js?template=' + template,
          async: 'async'
        })
      );
    }
  }]);

  return CodeFund;
}(_react2.default.Component);

exports.default = CodeFund;


CodeFund.defaultProps = {
  template: 'default'
};
