// redux-form-website-template/es/components/Home.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _GithubButton = require('./GithubButton');

var _GithubButton2 = _interopRequireDefault(_GithubButton);

var _CodeFund = require('./CodeFund');

var _CodeFund2 = _interopRequireDefault(_CodeFund);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Home = function Home(_ref) {
  var version = _ref.version;

  var styles = require('./Home.scss');
  return _react2.default.createElement(
    'div',
    { className: styles.home },
    _react2.default.createElement(
      'div',
      { className: styles.masthead },
      _react2.default.createElement('div', { className: styles.logo }),
      _react2.default.createElement(
        'h1',
        null,
        'Redux Form'
      ),
      _react2.default.createElement(
        'div',
        { className: styles.version },
        'v',
        version
      ),
      _react2.default.createElement(
        'h2',
        null,
        'The best way to manage your form state in Redux.'
      ),
      _react2.default.createElement(_GithubButton2.default, {
        user: 'erikras',
        repo: 'redux-form',
        type: 'star',
        width: 160,
        height: 30,
        count: true,
        large: true
      }),
      _react2.default.createElement(_GithubButton2.default, {
        user: 'erikras',
        repo: 'redux-form',
        type: 'fork',
        width: 160,
        height: 30,
        count: true,
        large: true
      }),
      _react2.default.createElement(
        'div',
        { style: { textAlign: 'center' } },
        _react2.default.createElement('hr', null),
        _react2.default.createElement(_CodeFund2.default, null)
      )
    ),
    _react2.default.createElement(
      'div',
      { className: styles.options },
      _react2.default.createElement(
        'a',
        { href: 'docs/GettingStarted.md' },
        _react2.default.createElement('i', { className: styles.start }),
        'Start Here'
      ),
      _react2.default.createElement(
        'a',
        { href: 'docs/api' },
        _react2.default.createElement('i', { className: styles.api }),
        'API'
      ),
      _react2.default.createElement(
        'a',
        { href: 'examples' },
        _react2.default.createElement('i', { className: styles.examples }),
        'Examples'
      ),
      _react2.default.createElement(
        'a',
        { href: 'docs/faq' },
        _react2.default.createElement('i', { className: styles.faq }),
        'FAQ'
      )
    )
  );
};

exports.default = Home;
