// redux-form-website-template/es/components/Sponsor.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _jsCookie = require('js-cookie');

var _jsCookie2 = _interopRequireDefault(_jsCookie);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var key = '$$$$$$$$';

function load() {
  var fromCookie = _jsCookie2.default.getJSON(key);
  return fromCookie ? Promise.resolve(fromCookie) : new Promise(function (resolve) {
    fetch('https://raw.githubusercontent.com/erikras/redux-form/master/docs/sponsor.json').then(function (response) {
      response.json().then(function (data) {
        resolve(data);
        _jsCookie2.default.set(key, data, {
          expires: 0.04 /* 1 hour */
          , path: '/'
        });
      });
    });
  });
}

var Sponsor = function (_React$Component) {
  _inherits(Sponsor, _React$Component);

  function Sponsor() {
    _classCallCheck(this, Sponsor);

    var _this = _possibleConstructorReturn(this, (Sponsor.__proto__ || Object.getPrototypeOf(Sponsor)).call(this));

    _this.state = { loaded: false };
    return _this;
  }

  _createClass(Sponsor, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      load().then(function (_ref) {
        var id = _ref.id,
            copy = _ref.copy;

        _this2.setState({
          loaded: true,
          id: id,
          copy: copy
        });
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          loaded = _state.loaded,
          id = _state.id,
          copy = _state.copy;

      return loaded ? _react2.default.createElement(
        'div',
        { className: 'benevolent-sponsor' },
        _react2.default.createElement('img', {
          src: 'https://codesponsor.io/t/l/' + id + '/pixel.png',
          style: {
            position: 'absolute',
            bottom: 0,
            right: 0
          },
          alt: 'pixel',
          width: '1',
          height: '1'
        }),
        _react2.default.createElement('a', {
          target: '_blank',
          href: 'https://codesponsor.io/t/c/' + id + '/',
          dangerouslySetInnerHTML: { __html: copy }
        })
      ) : _react2.default.createElement('div', null);
    }
  }]);

  return Sponsor;
}(_react2.default.Component);

exports.default = Sponsor;
